github3/session.py

```diff
@@ -110,6 +110,10 @@ def oauth2_auth(self, client_id, client_secret):
         raise NotImplementedError('These features are not implemented yet')
 
     def request(self, *args, **kwargs):
+        if 'headers' in kwargs:
+            headers = kwargs.pop('headers')
+            self.headers.update(headers)
```

Comment:
I think I'm missing something. Can you explain why this is necessary?

Reply:
Sure thing. To both POST and GET label descriptions from the API, you must pass `'application/vnd.github.symmetra-preview+json'` as Accept header ([docs](https://developer.github.com/v3/issues/labels/#create-a-label)).

Based on current design, the Accept header is [set at init time in the session](https://github.com/sigmavirus24/github3.py/blob/a084f4ad6ee195827b2e2331f2e64792f1dc92a3/github3/session.py#L56). With this simple change, each method can customize it.

Maybe there's another way I'm not seeing?

![image](https://user-images.githubusercontent.com/872296/39976144-153aec78-5700-11e8-910c-ac08b3a41fca.png)

Reply:
This is excellent feedback. I think when I wrote this class that I was assuming everyone understood how python-requests works (which we're just ever-so-slightly altering). The "magic" is that Requests allows you to have "default" headers set on a Session and then override some or all of them through a request method. If we modify the "default" headers here they become the default for the rest of the Session's life and we don't want that. We only want to send our specific headers for these specific calls. Does that make sense?

Reply:
> I've worked with sessions before, I don't know how I missed this one. 

I almost missed this myself! I'm half-dead from PyCon so I was pretty convinced that there was some critical bug I was missing. :)

>  I'll close this one now, fix, and re-submit.

You can just push new commits to this same branch. No need to create a new PR. But feel free to use whatever workflow you're most comfortable with and I'm happy to work with you on that since you're so generously spending your time to fix this gap. :tada: